src/silo/query_engine/actions/details.cpp

```diff
@@ -2,22 +2,89 @@
 
 #include <nlohmann/json.hpp>
 
+#include "silo/common/date.h"
+#include "silo/common/string.h"
+#include "silo/config/database_config.h"
+#include "silo/database.h"
 #include "silo/query_engine/operator_result.h"
 #include "silo/query_engine/query_result.h"
+#include "silo/storage/column/date_column.h"
+#include "silo/storage/column/float_column.h"
+#include "silo/storage/column/indexed_string_column.h"
+#include "silo/storage/column/int_column.h"
+#include "silo/storage/column/pango_lineage_column.h"
+#include "silo/storage/column/string_column.h"
+#include "silo/storage/database_partition.h"
 
 namespace silo::query_engine::actions {
+Details::Details(std::vector<std::string> fields)
+    : fields(std::move(fields)) {}
 
-Details::Details() = default;
+std::vector<config::DatabaseMetadata> parseFields(
+   const silo::Database& database,
+   const std::vector<std::string>& fields
+) {
+   if (fields.size() == 0) {
```

Comment:
Clang-Tidy: The 'empty' method should be used to check for emptiness instead of 'size'